smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpConnector.java

```diff
@@ -219,7 +219,10 @@ void setup(ExecutionHolder executionHolder) {
 
         return ReactiveStreams.<Message<?>> builder()
                 .via(processor)
-                .onError(t -> opened.put(oc.getChannel(), false))
+                .onError(t -> {
+                    log.failureReported(oc.getChannel(), t);
+                    opened.put(oc.getChannel(), false);
+                })
```

Comment:
I have contemplated calling `reportFailure(oc.getChannel(), t)`, which is defined as:

```java
public void reportFailure(String channel, Throwable reason) {
    log.failureReported(channel, reason);
    opened.put(channel, false);
    terminate(null);
}
```

However, the `terminate` method seems to close all channels managed by this connector. I can only see one reason to do that: application shutdown (and the `terminate` method already observes the corresponding CDI event). There's one caller of `reportFailure`, which I don't pretend to understand.

So I left the `reportFailure` method alone and did just this :-)

Reply:
In Reactive Streams, failures are terminal. 
But I would add a "fatal" flag (as in the Kafka connector), to avoid stopping everything if we can recover.

Reply:
Yea I understand that reactive streams are always `onItem* (onComplete|onError)`. What I find weird is that we have a piece of code (`AmqpFailStop.handle`) calling `terminate` (indirectly, through `AmqpConnector.reportFailure`), which terminates all running streams in the connector (if I understand it correctly). I'm sure there was a reason, I just don't see it :-)